### DISCUSSION TITLE
Which controller sets providerID on bootstrapped node

### DISCUSSION THREAD

**Discussion by aohoyd** [kubernetes-sigs/cluster-api#7058] [Q&A] [answered]
Hey everyone, can someone forward me to the right direction?
I'm trying to figure out when `providerID` property is set in cluster-api kubeadm bootstrap for a node. I didn't find any clue in capi sources and I have no idea where to look for that. Do kubeadm or kubelet create it?

**Comment by aohoyd** [comment, marked as answer]
Looks like it should be done by cloud-controller-manager, I think I'm done here :)

**Reply by sbueringer** [reply]
That's correct!
